crates/router/src/core/payments.rs

```diff
@@ -4445,7 +4476,43 @@ where
     };
     Ok(connector)
 }
+#[cfg(feature = "v2")]
+#[allow(clippy::too_many_arguments)]
+pub async fn get_connector_choice_for_sdk_session_token<F, Req, D>(
+    operation: &BoxedOperation<'_, F, Req, D>,
+    state: &SessionState,
+    req: &Req,
+    merchant_account: &domain::MerchantAccount,
+    _business_profile: &domain::Profile,
+    key_store: &domain::MerchantKeyStore,
+    payment_data: &mut D,
```

Comment:
is mutable reference required here?

Reply:
This function is not needed anymore. So removed it.